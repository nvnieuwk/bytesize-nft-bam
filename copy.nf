process COPY {
    input:
    tuple val(meta), path(input)

    output:
    tuple val(meta), path("*.${input.extension}"), emit: copy

    script:
    """
    cp $input copy.${input.extension}
    """
}