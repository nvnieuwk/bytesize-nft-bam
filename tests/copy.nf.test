nextflow_process {

    name "Test Process COPY"
    script "copy.nf"
    process "COPY"

    test("BAM") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [id:"bam"],
                    file("${baseDir}/data/test.paired_end.bam")
                ])
                """
            }
        }

        then {
            assert snapshot(process.out.copy.collect { file(it[1]).name }).match()
        }

    }

    test("CRAM") {

        when {
            process {
                """
                input[0] = Channel.of([
                    [id:"cram"],
                    file("${baseDir}/data/test.paired_end.cram")
                ])
                """
            }
        }

        then {
            assert snapshot(process.out.copy.collect { file(it[1]).name }).match()
        }

    }

}
