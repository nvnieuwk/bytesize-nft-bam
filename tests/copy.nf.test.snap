{
    "CRAM": {
        "content": [
            [
                "copy.cram"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-30T10:23:49.511360935"
    },
    "BAM": {
        "content": [
            [
                "copy.bam"
            ]
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-30T10:23:43.591449039"
    }
}